jobs/build-kola-containers.Jenkinsfile

```
@@ -0,0 +1,224 @@
+            userRemoteConfigs: [[url: params.COREOS_ASSEMBLER_GIT_URL]],
+            extensions: [[$class: 'CloneOption',
+                          noTags: true,
+                          reference: '',
+                          shallow: true]]
+        ]
+    )
+
+    gitref = params.COREOS_ASSEMBLER_GIT_REF
+    def output = shwrapCapture("git rev-parse HEAD")
+    commit = output.substring(0,40)
+    shortcommit = commit.substring(0,7)
+}
+
+node {
+    def path=params.PATH_TO_CONTEXTS
+    // gather the context folders list
+    contexts = sh(returnStdout: true, script: """
+        cd ${path}
+        find . -maxdepth 1 -mindepth 1 -type d -not -path '*/\\.*' -exec basename {} \\;
+    """).trim().split("\n")
+}
+
+currentBuild.description = "[${gitref}@${shortcommit}] Waiting"
+
+// Get the list of requested architectures to build for
+def basearches = params.ARCHES.split() as Set
+// and  the list of contexts to build
+def contexts = contexts as Set
+
+lock(resource: "build-containers") {
+    cosaPod(image: "quay.io/coreos-assembler/coreos-assembler",
+            memory: "512Mi", kvm: false,
+            serviceAccount: "jenkins") {
+    timeout(time: 60, unit: 'MINUTES') {
+    try {
+
+        currentBuild.description = "[${gitref}@${shortcommit}] Running"
+
+        // By default we will allow re-using cache layers for one day.
+        // This is mostly so we can prevent re-downloading the RPMS
+        // and repo metadata and over again in a given day for successive
+        // builds.
+        def cacheTTL = "24h"
+        def force = ""
+        if (params.FORCE) {
+            force = '--force'
+            // Also set cacheTTL to 0.1s to allow users an escape hatch
+            // to force no cache layer usage.
+            cacheTTL = "0.1s"
+        }
+
+        withCredentials([file(credentialsId: 'jbtrystram-testorg-robot-push-secret', variable: 'REGISTRY_SECRET')]) {
+            stage('Build Containers') {
+                parallel basearches.collectEntries{arch -> [arch, {
+                    for (directory in contexts) {
+                        def dir = params.PATH_TO_CONTEXTS + directory
+                        pipeutils.withPodmanRemoteArchBuilder(arch: arch) {
+                            shwrap("""
+                                cosa remote-build-container \
+                                    --git-sub-dir $dir\
+                                    --arch $arch --cache-ttl ${cacheTTL} \
+                                    --git-ref $commit ${force} \
+                                    --git-url ${params.COREOS_ASSEMBLER_GIT_URL} \
+                                    --repo ${params.CONTAINER_REGISTRY_ORG}/$directory \
+                                    --push-to-registry --auth=\$REGISTRY_SECRET
+                            """)
+                        }
+                }
+                }]}
+            }
+
+            stage('Push manifests') {
```

Super minor, but for consistency

```suggestion
            stage('Push Manifests') {
```